provides.py

```diff
@@ -96,6 +96,15 @@ def sign_auth_request(self, scope, user, kubelet_token, proxy_token,
                 'creds': all_creds
             })
 
+    def clear_creds(self):
+        """
+        Clear creds from the relation. This is used by non-leader units to stop
+        advertising creds so that the leader can assume full control of them.
+        """
+        DB.unset('creds')
+        for relation in self.relations:
+            relation.to_publish_raw['creds'] = ''
```

Comment:
It seems like this is going to leave the `creds` key with an empty value.  Don't we want that key removed entirely from follower relation data so that only lead values matter?

Reply:
In my testing, this resulted in the creds key being removed completely from the relation data.

Behind the scenes this becomes a call to:
```
relation-set -r kube-control:X creds=
```

which removes the key ([source](https://juju.is/docs/charm-writing/hook-tools)):

> relation-set writes the local unit’s settings for some relation ... Setting an empty string causes the setting to be removed.

I could not find a cleaner way to do this. You can't do it with `to_publish` because the empty string will serialize to a non-empty string, and you can't do it using `del` because reactive won't call relation-set at all if the key is not in the dict. Seems like it has to be `to_publish_raw[key] = ''`.

Reply:
In that case, LGTM